packages/cdk8s/lib/app.ts

```diff
@@ -32,8 +32,13 @@ export class App extends Construct {
     this.outdir = options.outdir ?? process.env.CDK8S_OUTDIR ?? 'dist';
   }
 
-  protected onPrepare() {
-    super.onPrepare()
+  /**
+   * Synthesizes all manifests to the output directory
+   */
+  public synth(): void {
+    if (!fs.existsSync(this.outdir)) {
+      fs.mkdirSync(this.outdir);
```

Comment:
Use `{ recurse: true }` instead of this `if`

Reply:
👍